### PR TITLE
Migrate view component specs to provided component format

### PR DESCRIPTION
**Pull request by peteryates**
### Context and changes

ViewComponent comes with [a set of testing helpers](https://viewcomponent.org/guide/testing.html) that we're not using. Instead we have our own set that offer roughly the same functionality. As it's one extra thing to navigate and maintain, let's use the provided ones.

We're using a home-rolled one that seems to have more or less the same functionality, we may as well use the one that comes with ViewComponent.

The main differences are:

#### `stub_component` is gone

Instead of stubbing we're letting the component render its slots and checking that the right content is there using plain CSS matchers. As the slots are tested independently we just need a vague check, like ensuring the right number of rows are present.

#### `component` is gone:

Now we're using `render_inline` and assigning the rendered component to a variable via `let` or `subject`

### Guidance to review

I've tried not to touch any of the components themselves but did have to make the following minor changes:

* replace calls to `t` with `I18n.t`
* stop relying on Rails magic for linking, instead add a couple of `path_ids` so we need to jump through fewer hoops to test. I'll highlight these in the PR with comments for extra attention.

### CODE REVIEW

File: app/components/schools/participants/coc_status_table_row.rb [hunk 1]
```diff
@@ -37,6 +37,10 @@ def date_column_value
         end
       end
 
+      def path_ids
```
Comment (peteryates) [inline app/components/schools/participants/coc_status_table_row.rb:40]: Here's a method I added so we're not relying on Rails already knowing the path to build links, it's more explicit and easier to test.

File: app/components/schools/participants/status_table_row.rb [hunk 2]
```diff
@@ -26,6 +26,10 @@ def participant_is_on_a_cip?
         profile.school_cohort.school_chose_cip?
       end
 
+      def path_ids
```
Comment (peteryates) [inline app/components/schools/participants/status_table_row.rb:29]: Here's another method I added so we're not relying on Rails already knowing the path to build links, it's more explicit and easier to test.

File: app/components/admin/schools/cohorts/cip_info.rb [hunk 3]
```diff
@@ -4,7 +4,10 @@ module Admin
   module Schools
     module Cohorts
       class CipInfo < BaseComponent
-        def initialize(school_cohort:)
+        attr_accessor :school
+
+        def initialize(school:, school_cohort:)
```
[the remark this answers is not in the document]
Comment (tonyheadford) [reply app/components/admin/schools/cohorts/cip_info.rb:9]: Can we not use `school_cohort.school` here instead or is this also fixing an issue?

**Review by tonyheadford — COMMENTED** [review]
Couple of comments but this looks great

File: app/components/admin/schools/cohorts/cip_info.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (peteryates) [reply app/components/admin/schools/cohorts/cip_info.rb:9]: Ah good call. It's only necessary for `OtherInfo`, I'll remove this one.

**Review by tonyheadford — APPROVED** [review]

**Pull request merged**